registry/storage/driver/gcs/gcs.go

```diff
@@ -864,33 +818,33 @@ func startSession(client *http.Client, bucket string, name string) (uri string,
 	return uri, err
 }
 
-func putChunk(client *http.Client, sessionURI string, chunk []byte, from int64, totalSize int64) (int64, error) {
-	bytesPut := int64(0)
-	err := retry(func() error {
-		req, err := http.NewRequest(http.MethodPut, sessionURI, bytes.NewReader(chunk))
-		if err != nil {
-			return err
-		}
-		length := int64(len(chunk))
-		to := from + length - 1
-		size := "*"
-		if totalSize >= 0 {
-			size = strconv.FormatInt(totalSize, 10)
-		}
-		req.Header.Set("Content-Type", "application/octet-stream")
-		if from == to+1 {
-			req.Header.Set("Content-Range", fmt.Sprintf("bytes */%v", size))
-		} else {
-			req.Header.Set("Content-Range", fmt.Sprintf("bytes %v-%v/%v", from, to, size))
-		}
-		req.Header.Set("Content-Length", strconv.FormatInt(length, 10))
+func (w *writer) putChunk(ctx context.Context, sessionURI string, chunk []byte, from int64, totalSize int64) (int64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, sessionURI, bytes.NewReader(chunk))
+	if err != nil {
+		return 0, err
+	}
+	length := int64(len(chunk))
+	to := from + length - 1
+	size := "*"
+	if totalSize >= 0 {
+		size = strconv.FormatInt(totalSize, 10)
+	}
+	req.Header.Set("Content-Type", blobContentType)
+	if from == to+1 {
+		req.Header.Set("Content-Range", fmt.Sprintf("bytes */%v", size))
```

Comment:
`size` is a string; could safe some cycles by using a straight concat here;

```go
req.Header.Set("Content-Range", "bytes *" + size)
```

Or otherwise make it explicit that it's a string;

```go
req.Header.Set("Content-Range", fmt.Sprintf("bytes */%s", size))
```

Reply:
done: https://github.com/distribution/distribution/pull/4120/commits/9b3043cbd4617855783215592423344b95049f3d